bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/GC.java

```diff
@@ -5120,4 +5158,12 @@ private static int sin(int angle, int length) {
 	return (int)(Math.sin(angle * (Math.PI/180)) * length);
 }
 
+private int getNativeDeviceZoom() {
+	return data.nativeZoom != 0 ? data.nativeZoom : DPIUtil.getNativeDeviceZoom();
```

Comment:
You are right. The implementation of extractZoom was put, later on and with the current implementation, we don't need getNativeZoom method since it makes sure that the GCData would always have a nativeZoom.